test.js

```diff
@@ -317,3 +317,16 @@ test('toStdout without factory', async function (t) {
 
   t.ok(completed)
 })
+
+test('should allow for awaiting the response with default stdout stream', async function (t) {
+  t.plan(1)
+
+  try {
+    await helpMe({
+      dir: 'fixture/basic'
+    }).toStdout([])
+    t.ok('awaited correctly')
```

Comment:
Mocking `process.stdout.write` is not trivial, as this is where test output goes, and doing so in the test is messing it up.
Imo making an assertion on correct await should be symmetrical to making an assertion on the stream being triggered (for this exact test), as, without it, it would be impossible.
If you have any ideas how to test that, I'm all ears :)

Reply:
Here is a patch to implement it:

```patch
diff --git a/help-me.js b/help-me.js
index eecfd6a..f730d7b 100644
--- a/help-me.js
+++ b/help-me.js
@@ -3,6 +3,7 @@
 const fs = require('fs')
 const { PassThrough, Writable, pipeline } = require('stream')
 const glob = require('glob')
+const process = require('process')
 
 const defaults = {
   ext: '.txt',
diff --git a/package.json b/package.json
index 84468c0..576edd4 100644
--- a/package.json
+++ b/package.json
@@ -28,6 +28,7 @@
     "pre-commit": "^1.1.3",
     "standard": "^16.0.0",
     "tap-spec": "^5.0.0",
+    "proxyquire": "^2.1.3",
     "tape": "^5.0.0"
   },
   "dependencies": {
diff --git a/test.js b/test.js
index 667b83e..c11a931 100644
--- a/test.js
+++ b/test.js
@@ -5,6 +5,7 @@ const concat = require('concat-stream')
 const fs = require('fs')
 const path = require('path')
 const helpMe = require('./')
+const proxyquire = require('proxyquire')
 
 test('throws if no directory is passed', function (t) {
   try {
@@ -321,12 +322,22 @@ test('toStdout without factory', async function (t) {
 test('should allow for awaiting the response with default stdout stream', async function (t) {
   t.plan(1)
 
-  try {
-    await helpMe({
-      dir: 'fixture/basic'
-    }).toStdout([])
-    t.ok('awaited correctly')
-  } catch {
-    t.fail('should not reject')
+  const _process = Object.create(process)
+  const stdout = Object.create(process.stdout)
+  Object.defineProperty(_process, 'stdout', {
+    value: stdout
+  })
+
+  stdout.write = (data, cb) => {
+    t.equal(data.toString(), 'hello world\n')
+    cb()
   }
+
+  const helpMe = proxyquire('./help-me', {
+    process: _process
+  })
+
+  await helpMe.help({
+    dir: 'fixture/basic'
+  }, ['help'])
 })
```

Reply:
TIL about `proxyquire`, that's neat, thanks! Updated